src/main/java/com/mathworks/ci/MatlabCommandStepExecution.java

```diff
@@ -43,10 +43,15 @@ public Void run() throws Exception {
         
         workspace.mkdirs();
         
-        int res = execMatlabCommand(workspace, launcher, listener, env);
+        int exitCode = execMatlabCommand(workspace, launcher, listener, env);
 
-        getContext().setResult((res == 0) ? Result.SUCCESS : Result.FAILURE);
-        
+        if(exitCode == 0){
```

Comment:
Perhaps it's a matter of style, but having the error condition in an if condition and returning feels more readable. Also, it allows you to add more code below the error condition in case you want to do more things before returning. 

if(exitCode !=0 {
 // throw an exception if return code is non-zero
        stop(new MatlabExecutionException(exitCode));
        return null;
}

getContext().setResult(Result.SUCCESS);
return null;